src/DataUtils.cpp

```diff
@@ -0,0 +1,778 @@
+#include "NAS2D/DataUtils.h"
+
+#include "NAS2D/StringUtils.h"
+
+#include <stdexcept>
+
+bool NAS2D::Dictionary::value_helper(const std::string& key, bool) const
+{
+	auto valueAsStr = [this, &key]() -> std::string {
+		try
+		{
+			return value<std::string>(key);
+		}
+		catch (...)
+		{
+			throw;
+		}
+	}(); // IILE
+	valueAsStr = NAS2D::trimWhitespace(NAS2D::toLowercase(valueAsStr));
+	if (valueAsStr == "true" || valueAsStr == "false")
+	{
+		return valueAsStr == "true";
+	}
+	else
+	{
+		const auto valueAsBool = [&valueAsStr]() {
```

Comment:
I suppose similar advice can be given concerning the bool format about being strict in terms of what you accept. Perhaps we only want to allow either `"true"` or `"false"` for boolean values, and not `0`, or `1`.

Reply:
I prefer being robust with boolean values. Especially since it's really only two checks, "If the value is convertible to a string, is the lowercased value equal to "true"? If not, is it not zero (i.e., "true")?

Trimming white space is me being paranoid because some strings returned from functions include surrounding white space and would return false for an otherwise true value. TinyXml2 will happily do this if you ask for the Element Text or Value:

Compare this:
```
<foo>true</foo>
```
which returns `"true"`.

While this:
```
<foo>
true
</foo>
```
returns
`"\ntrue\n"`

Annoyances like above have made me use attributes exclusively when intending for strict text values.

Reply:
True... (pun?), though keep in mind these files will be machine written. The format should always be consistent. That's true of both spacing and casing.

----

I had to go looking to dig up the terms for what we're discussing. It's known as the [Robustness principle](https://en.wikipedia.org/wiki/Robustness_principle). You're proposing we follow the Robustness principle. I'm bringing up the objections from the Criticism section, which suggests the principle may actually harm robustness.

Another source on the topic is:
[The Harmful Consequences of the Robustness Principle](https://tools.ietf.org/id/draft-thomson-postel-was-wrong-03.html)

----

Oh, and we are exclusively using attributes for the values.